index.js

```diff
@@ -1 +1,83 @@
 // implement your API here
+const express = require('express');
+
+const db = require('./data/db.js');
+
+const server = express();
+
+server.use(express.json()); //makes post and put work
+
+server.get('/', (req, res) => {
+    res.send('Hello World!');
+})
+
+server.post('/api/users', (req, res) => {
+    const { name, bio } = req.body;
+    if(!name || !bio){
+      return res.status(400).json({ message: "Please provide name and bio for the user." })
+    }
+    db.insert({ name, bio }).then( user => {
+        res.status(201).json(user)
+    }).catch(error => {
+        res.status(500).json({ message: 'There was an error while saving the user to the database.' })
+    })
+})
+
+server.get('/api/users', (req, res) => {
+    db.find().then( users => {
+        res.status(200).json(users);
+    } ).catch( error => {
+        res.status(500).json({ message: 'The users information could not be retrieved.' })
+    })
+})
+
+server.get('/api/users/:id', (req, res) => {
+    const id = req.params.id;
+    db.findById(id).then(user => {
+      if(!user){
+        return res.status(404).json({ message: 'The user with the specified ID does not exist.' })
+      } else {
+        return res.status(200).json(user)
+      }
+    })
+    .catch( error => {
+        res.status(500).json({ message: 'The user information could not be retrieved.' })
+    })
+})
+
+server.delete('/api/users/:id', (req, res) => {
+    const id = req.params.id;
+    db.remove(id).then(deleted => {
+      if(!deleted){
+        return res.status(404).json({ message: "The user with the specified ID does not exist." })
+      } 
+        return res.status(204).end();
+    })
+    .catch(error => {
+      res.status(500).json({ message: 'The user could not be removed.' })
+  })
+})
+
+server.put('/api/users/:id', (req, res) => {
+    const id = req.params.id;
+    const changes = req.body;
+    const { name, bio } = req.body;
+
+    db.update(id, changes).then( updated => {
+      if(!updated){
+        res.status(404).json({ message: "The user with the specified ID does not exist." })
+      } else if (!name || !bio) {
+        res.status(400).json({ message: 'Please provide name and bio for the user.' })
```

Comment:
This conditional should go before db.update. If there's no name or bio, we shouldn't try and update.